catalog/templates/catalog/product_list.html

```diff
@@ -15,7 +15,7 @@ <h1 class="my-0 font-weight-normal">{{ product.name }}</h1>
                 <div class="card-body">
                     <h2 class="card-title pricing-card-title">{{ product.price }} руб.</h2>
                     <ul class="list-unstyled mt-3 mb-4 text-start m-3">
-                        <li>Категория: {{ product.category }}</li>
+                        Категория: <a href="{% url 'catalog:category_product' product.category.id %}">{{ product.category.name }}</a>
```

Comment:
Так будет оптимальнее - не нужно будет делать доп запрос за категорией в базу данных
```suggestion
                        Категория: <a href="{% url 'catalog:category_product' product.category_id %}">{{ product.category.name }}</a>
```